examples/complex/test/AppManager.test.js

```diff
@@ -0,0 +1,105 @@
+const OwnedUpgradeabilityProxy = artifacts.require('zos-lib/contracts/upgradeability/OwnedUpgradeabilityProxy.sol');
+const UpgradeabilityProxyFactory = artifacts.require('UpgradeabilityProxyFactory');
+const Package = artifacts.require('Package');
+const AppDirectory = artifacts.require('AppDirectory');
+const ContractDirectory = artifacts.require('ContractDirectory');
+const AppManager = artifacts.require('PackagedAppManager');
+const MintableERC721Token = artifacts.require('MintableERC721Token');
+
+const DonationsV1 = artifacts.require('DonationsV1');
+const DonationsV2 = artifacts.require('DonationsV2');
+
+const decodeLogs = require('zos-lib').decodeLogs;
+const encodeCall = require('zos-lib').encodeCall;
+const validateAddress = require('./helpers/validateAddress.js');
+const shouldBehaveLikeDonations = require('./Donations.behavior.js');
+const shouldBehaveLikeDonationsWithTokens = require('./DonationsWithTokens.behavior.js');
+const should = require('chai').should();
+const deploy = require('../index.js');
+
+contract('AppManager', (accounts) => {
+
+  before(function() {
+    this.owner = accounts[1];
+    this.donor1 = accounts[2];
+    this.wallet = accounts[4];
+    this.initialVersion = '0.0.1';
+    this.updatedVersion = '0.0.2';
+    this.contractName = "Donations";
+    this.tokenClass = 'MintableERC721Token';
+    this.tokenName = 'DonationToken';
+    this.tokenSymbol = 'DON';
+    this.txParams = {
+      from: this.owner
+    };
```

Comment:
If these variables are constants, I'd prefer defining them as such

Reply:
They are constants, but I'm using "this" because I need them in the context of the test.